pkg/providers/tinkerbell/upgrade.go

```diff
@@ -238,3 +275,33 @@ func (p *Provider) isScaleUpDown(currentSpec *cluster.Spec, newSpec *cluster.Spe
 
 	return false
 }
+
+func (p *Provider) validateMachineConfigImmutability(ctx context.Context, cluster *types.Cluster, newConfig *v1alpha1.TinkerbellMachineConfig, clusterSpec *cluster.Spec) error {
```

Comment:
Can we also check the `HardwareSelector`? I think that would also cause a rolling update because its part of the template-cp.